### PR TITLE
SED-2939 anonymous-user-wrongly-created-in-step-ee

### PR DESCRIPTION
**Pull request by david-stephan**

### CODE REVIEW

**Review by cl-exense — CHANGES_REQUESTED** [review]
Nice code! Just minor comments and quick things from my side.

File: step-controller/step-controller-server/src/main/java/step/core/deployment/NoAuthenticationFilter.java [hunk 1]
```diff
@@ -49,25 +54,27 @@ public void init() throws Exception {
...
         }
-        return session;
+    }
+
+    private synchronized User createAnonymousUserIfRequired() {
+        User anonymous = userAccessor.getByUsername("anonymous");
+        if (anonymous == null) {
+            anonymous = new User();
+            anonymous.setUsername("anonymous");
+            anonymous.setRole("admin");
```
[the remark this answers is not in the document]
Comment (david-stephan) [reply step-controller/step-controller-server/src/main/java/step/core/deployment/NoAuthenticationFilter.java]: @cl-exense not sure what was exactly the question, the change to synchronized was necessary and a low level of concurrency is expected. Still the old code systematically accessing the DB was not required, I just refactored it.

**Comment by david-stephan** [comment]
@cl-exense Thanks for the review, I think I addressed all points, let me know in there is anything else/new.

File: step-controller/step-controller-server/src/main/java/step/core/deployment/NoAuthenticationFilter.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (cl-exense) [reply step-controller/step-controller-server/src/main/java/step/core/deployment/NoAuthenticationFilter.java]: The question was about the setRole(admin) specifically, but nice catch with the DB refactoring.

**Review by cl-exense — APPROVED** [review]

**Pull request merged**